lib/manager/npm/update/dependency/index.ts

```diff
@@ -1,8 +1,53 @@
 import { dequal } from 'dequal';
+import type { PackageJson } from 'type-fest';
 import { logger } from '../../../../logger';
 import { matchAt, replaceAt } from '../../../../util/string';
 import type { UpdateDependencyConfig } from '../../../types';
 
+function replaceAsString(
+  parsedContents: PackageJson,
+  fileContent: string,
+  depType: string,
+  depName: string,
+  oldVersion: string,
+  newValue: string
+): string | null {
+  // Update the file = this is what we want
+  if (depType === 'resolutions') {
+    // eslint-disable-next-line no-param-reassign
+    parsedContents.resolutions[depName] = newValue;
+  } else {
+    // eslint-disable-next-line no-param-reassign
+    parsedContents[depType][depName] = newValue;
+  }
+  // Look for the old version number
+  const searchString = `"${oldVersion}"`;
+  const newString = `"${newValue}"`;
+  // Skip ahead to depType section
+  let searchIndex = fileContent.indexOf(`"${depType}"`) + depType.length;
+  logger.trace(`Starting search at index ${searchIndex}`);
+  // Iterate through the rest of the file
+  for (; searchIndex < fileContent.length; searchIndex += 1) {
+    // First check if we have a hit for the old version
+    if (matchAt(fileContent, searchIndex, searchString)) {
+      logger.trace(`Found match at index ${searchIndex}`);
+      // Now test if the result matches
+      const testContent = replaceAt(
+        fileContent,
+        searchIndex,
+        searchString,
+        newString
+      );
+      // Compare the parsed JSON structure of old and new
+      if (dequal(parsedContents, JSON.parse(testContent))) {
+        return testContent;
+      }
+    }
+  }
+  // istanbul ignore next: not possible to get here
+  return null;
```

Comment:
Should we throw an error Here? 🤔

Reply:
Good question. If I am reading the old code path right, if we got here, we would silently ignore it. Right?